activerecord/lib/active_record/core.rb

```diff
@@ -512,14 +510,8 @@ def inspect
       # We check defined?(@attributes) not to issue warnings if the object is
       # allocated but not initialized.
       inspection = if defined?(@attributes) && @attributes
-        self.class.attribute_names.collect do |name|
-          if has_attribute?(name)
-            if filter_attribute?(name)
-              "#{name}: #{ActiveRecord::Core::FILTERED}"
-            else
-              "#{name}: #{attribute_for_inspect(name)}"
-            end
-          end
+        attributes_for_inspect(self.class.filter_attributes).map do |name, value|
+          "#{name}: #{value}"
```

Comment:
Only `#inspect` wants truncated strings and formatted Date values, so perhaps it should be calling that here, rather than expecting `attributes_for_inspect` to provide them:
```ruby
attributes_for_inspect.map do |name, value|
  "#{name}: #{attribute_value_for_inspect(value)}"
end.join(", ")
```